modules/programs/numbat.nix

```
@@ -46,5 +63,16 @@ in
     home.file."${configDir}/config.toml" = mkIf (cfg.settings != { }) {
       source = tomlFormat.generate "numbat-config" cfg.settings;
     };
+
+    home.file."${configDir}/init.nbt" = mkIf (cfg.initFile != null) (
+      if (cfg.initFile.source != null) then
+        {
+          source = cfg.initFile.source;
+        }
+      else
+        {
+          text = cfg.initFile.text;
+        }
+    );
```

this seems much simpler, if there isn't any reason not to.
```suggestion
    home.file."${configDir}/init.nbt" = mkIf (cfg.initFile != null) cfg.initFile;
```

---

This is a really nice simplication, but it doesn't seem to work as [sourceFileOrLines sets source to null](https://github.com/nix-community/home-manager/blob/c7fdb7e90bff1a51b79c1eed458fb39e6649a82a/modules/lib/types.nix#L231). Since null is not a valid type for an absolute path, this causes a couple tests to fail [here](https://github.com/AyushmaanAggarwal/home-manager/blob/bcea8b77b6deaaf4c1420ae6079a4a5c1f5d2257/tests/modules/programs/numbat/example-init-text.nix)